packages/langium/src/syntax-tree.ts

```diff
@@ -240,3 +240,23 @@ export interface RootCstNode extends CompositeCstNode {
 export function isRootCstNode(node: unknown): node is RootCstNode {
     return isCompositeCstNode(node) && 'fullText' in node;
 }
+
+//Manipulates a type to have only properties whose value is of a certain type.
+type ExtractKeysOfValueType<T, K> = { [I in keyof T]: T[I] extends K ? I : never }[keyof T];
+
+/**
+ * Returns the property names (!) of an AstNode that are cross-references.
+ * Meant to be used during cross-reference resolution in combination with `assertUnreachable(context.property)`.
+ */
+export type CrossReferencesOfAstNodeType<N extends AstNode> = (
+    ExtractKeysOfValueType<N, Reference|undefined>
+    | ExtractKeysOfValueType<N, Array<Reference|undefined>|undefined>
+) & Record<string, never>;
+
+/**
+ * Returns all types that contain cross-references, A is meant to be the interface `XXXAstType` fromm your generated `ast.ts` file.
```

Comment:
We have the type `AstTypeList` defined in the same file for this. Proposal: move that type right before these new definitions and add a JSDoc comment to it.

Reply:
Nice! Done